TrafficCapture/trafficReplayer/src/testFixtures/java/org/opensearch/migrations/replay/RootReplayerConstructorExtensions.java

```diff
@@ -0,0 +1,69 @@
+package org.opensearch.migrations.replay;
+
+import org.opensearch.migrations.replay.tracing.IRootReplayerContext;
+import org.opensearch.migrations.replay.traffic.source.TrafficStreamLimiter;
+import org.opensearch.migrations.transform.IAuthTransformerFactory;
+import org.opensearch.migrations.transform.IJsonTransformer;
+
+import javax.net.ssl.SSLException;
+import java.net.URI;
+import java.time.Duration;
+
+public class RootReplayerConstructorExtensions extends TrafficReplayerTopLevel {
```

Comment:
Would the builder pattern help here simplify the constructor behavior with optional and default parameters?

Reply:
I've tried the builder pattern a couple times.  One big issue is that it's a lot of code to maintain (even more than what's here).   Lombok won't work since I want to set defaults.  For now, I would prefer to have to think about and see all of the arguments rather than forget about one.  If this code gets to the point where every combination is in a constructor, then the builder pattern would obviously be better.  For now though, I don't like how with the builder pattern, I could, for example, forget to specify the URI.  Sure, that could be a runtime check, but this way can do compile time checks.

Of course, I could so something like a hybrid builder where the builder has a c'tor and then setters (or composed parts).  I don't know if anybody would like that and it seems burdensome.  I was comforted a bit when I had realized that this is only required because of testing.